### PR TITLE
#155 bug fixed

### PR DESCRIPTION
**Pull request by yuvalron**
I don't know how to fix this without using subscription. In order to get the faculty details, I first need the faculty-id from the course details, the problem is that the course itself is already observable.

### CODE REVIEW

**Comment by ron4548** [comment]
Use pipe with flatMap. You can see many examples for such usage in topic and question components.

File: src/app/ui/upload/upload.component.ts [hunk 1]
```diff
@@ -84,7 +86,10 @@ export class UploadComponent implements OnInit {
       this.semester = parseInt(split[1].substr(5, 2), 10);
       const moedId = parseInt(split[3], 10);
       this.moed = (moedId === 1) ? 'A' : (moedId === 2) ? 'B' : 'C';
-      this.db.getCourse(courseId).subscribe(course => this.course = course);
+      this.db.getCourse(courseId).subscribe(course => {
+          this.course = course;
+          this.db.getFaculty(course.faculty).subscribe(faculty => this.faculty = faculty);
```
Comment (ron4548) [inline src/app/ui/upload/upload.component.ts]: It's not good. Every time there is new value you make a subscription to the inner observable. It results with memory leak.

Comment (yuvalron) [reply src/app/ui/upload/upload.component.ts]: What to do here? 
This code is in the upload.component.ts file
image

Comment (yuvalron) [reply src/app/ui/upload/upload.component.ts]: I already changed course to course$ which is an Observable<Course>

Comment (ron4548) [reply src/app/ui/upload/upload.component.ts]: maybe convert these observables to Promise? see if something like this works out:
course$.pipe(take(1)).toPromise().then(course => { do stuff});

Comment (yuvalron) [reply src/app/ui/upload/upload.component.ts]: @ron4548 
tried this but the upload doesn't work :/
image

Comment (ron4548) [reply src/app/ui/upload/upload.component.ts]: you need to return the promise returned by uploadScan

**Pull request merged**